app/models/concerns/feature_flaggable.rb

```diff
@@ -12,6 +12,42 @@ def feature_flags_with_defaults
     FeatureFlag.default_flag_hash.merge(self.feature_flags ? self.feature_flags : {}).with_indifferent_access
   end
 
+  # updates the feature_flags on the instance to the updated_flag_hash,
+  # with saftey checks to ensure values are valid and allowed
+  # suitable for updating flags from user input.
+  # throws an error if the update fails
+  # returns the value from the update! operation
+  def update_feature_flags_safe!(updated_flag_hash, allowed_flag_names)
```

Comment:
I think this should be implemented as a validation rather than a special save function.  It will make the controller logic much simpler.